test/clojure_tic_tac_toe/game_setup_test.clj

```diff
@@ -0,0 +1,43 @@
+(ns clojure-tic-tac-toe.game_setup_test
+  (:require [clojure.string :as str]
+            [clojure.test :refer [deftest testing is]]
+            [clojure-tic-tac-toe.game_setup :refer :all]))
+
+(defn- set-up-game-output []
+  (with-out-str
+    (with-in-str "h\n"
+      (set-up-game))))
+
+(deftest set-up-game-test
+  (testing "when the player chooses to play another person"
+    (is (= true
+           (str/includes?
+             (set-up-game-output)
+             "Tic Tac Toe"))
```

Comment:
another way to test this might be to use a mock function (I've seen people using `with-redefs`) to check that the functions from io were called instead of checking the actual strings. You'll see that a lot of your tests here are pretty redundant from the ones in your io tests.

This may also apply up in your console_ui when you're testing the game setup. I didn't leave a comment up there since you may have intended it to be more of an integration level test.

I'm not sure if you've seen this talk but it touches on this topic: https://www.destroyallsoftware.com/talks/boundaries

Reply:
I agree that using `with-redefs` here would be a good way to clean up my tests. I am not exactly sure how to use them though, and plan on talking with @jdesrosiers sometime about them.